src/Microsoft.AspNetCore.Mvc.Core/Internal/DefaultCollectionValidationStrategy.cs

```diff
@@ -45,6 +46,7 @@ public class DefaultCollectionValidationStrategy : IValidationStrategy
         /// Gets an instance of <see cref="DefaultCollectionValidationStrategy"/>.
         /// </summary>
         public static readonly IValidationStrategy Instance = new DefaultCollectionValidationStrategy();
+        private static readonly Dictionary<Type, MethodInfo> _genericGetEnumeratorCache = new Dictionary<Type, MethodInfo>();
```

Comment:
- Why is this `static` vs. a field inside the single `static` instance of `DefaultCollectionValidationStrategy`?
- Suggest this needs to be at least a `ConcurrentDictionary` before it can be shared across threads.